### ISSUE TITLE
Feature Request/Idea: Include Grant ID in "Schema.org JSON-LD" metadata export format

### ISSUE THREAD

**Issue by kbrueckmann** [IQSS/dataverse#10296]
**Overview of the Feature Request**
The proposal of this feature is to include the grant number in the metadata export format "Schema.org JSON-LD". So far only the funding agency is included in this format (as [funder](https://schema.org/funder)), whereas in others like OpenAIRE the grant number is also given (see Dataverse metadata crosswalk). The idea was shortly mentioned in the [Dataverse chat](https://dataverse.zulipchat.com/#narrow/stream/378866-troubleshooting/topic/metadata.20mapping.20in.20schema.2Eorg.20json-ld.20format) to @pdurbin. 
As far as I can see the Schema.org property [funding](https://schema.org/funding) might be quite well-suited for this.

**What kind of user is the feature intended for?**
API User

**What inspired the request?**
A comparison of information given via the metadata input fields with the amount of information received from the metadata export formats.

**What existing behavior do you want changed?**
Expanding the Schema.org JSON-LD metadata export format.

**Any open or closed issues related to this feature request?**
None found with a search. #4859 might be broadly related.

**Comment by jggautier** [comment]
Thanks @kbrueckmann! It's great that you've brought this up, since we're doing research (https://github.com/IQSS/dataverse/issues/4859 and https://github.com/IQSS/dataverse/issues/10196) that will likely lead to changes in how Dataverse organizes funding metadata, which will affect how fields are mapped to Schema.org properties and other metadata exports.

Right now I'd recommend planning to change the Schema.org export to include grant numbers based on how Dataverse currently organizes funding metadata. Then we can revisit the mappings when we know more about how we'll re-organize Dataverse's funding metadata fields.

It was mostly [Google's recommendations](https://developers.google.com/search/docs/appearance/structured-data/dataset) that influenced how Dataverse exports metadata using Schema.org. We talked a lot about those recommendations in https://github.com/IQSS/dataverse/issues/2243. Google's recommendations still don't mention how to include grant numbers. So back in 2020, [I wrote](https://github.com/IQSS/dataverse/issues/4371#issue-280615933) that there wasn't a way to include things like grant numbers.

But the following year when [we were improving the Schema.org export](https://github.com/IQSS/dataverse/issues/7349#issuecomment-780066428), we mentioned guidelines from others that included ways to add things like grant numbers, namely guidelines from "Science on Schema.org": https://github.com/ESIPFed/science-on-schema.org/blob/master/guides/Dataset.md#funding.

I think it'll be important to consider how Google's validation tools will respond to changes in the Schema.org export that include grant numbers. Several folks who manage different Dataverse repositories keep an eye on the Google Search Console's reports on the validity of the Schema.org in their dataset pages. Will the Google Search Console start complaining if funding information is included the way that OpenAIRE does it or how it's recommended from the guidelines from "Science on Schema.org" guidelines? I think this could be tested using Google's validation tools.

**Comment by cmbz** [comment]
2025-06-30
- @jggautier will review in light of new work on GREI metadata and grant information and decide to either close the issue or move it forward.

**Comment by jggautier** [comment]
Hmm, I don't know of anyone involved in the NIH GREI grant-related work talking about how grant IDs aren't included in Schema.org metadata exports. I think the Croissant metadata export is based on Schema.org, and in our NIH GREI reports we've talked about Croissant.

But I think the focus for the NIH GREI related work has mostly been on what repositories send to DataCite using it's metadata schema, and signposting.

I think being able to include grant IDs in Schema.org exports might be helpful and seems possible, but I don't think this is a priority.

@kbrueckmann, could you share how much of a priority this is for you?

And what do you think about [what I wrote in Feb](https://github.com/IQSS/dataverse/issues/10296#issuecomment-1927145158)?:

> But the following year when we were improving the Schema.org export, we mentioned guidelines from others that included ways to add things like grant numbers, namely guidelines from "Science on Schema.org": https://github.com/ESIPFed/science-on-schema.org/blob/master/guides/Dataset.md#funding.
> 
> I think it'll be important to consider how Google's validation tools will respond to changes in the Schema.org export that include grant numbers. Several folks who manage different Dataverse repositories keep an eye on the Google Search Console's reports on the validity of the Schema.org in their dataset pages. Will the Google Search Console start complaining if funding information is included the way that OpenAIRE does it or how it's recommended from the guidelines from "Science on Schema.org" guidelines? I think this could be tested using Google's validation tools.

- Do you think it would be possible for you or someone in the Dataverse community to follow those guidelines from "Science on Schema.org" so that grant IDs are included?
- Do you think Google's validation tools would be okay with those changes?

**Comment by pdurbin** [comment]
> I think the Croissant metadata export is based on Schema.org

Yes, definitely. "Croissant builds on the [schema.org/Dataset](http://schema.org/Dataset) vocabulary, which is widely adopted by datasets on the web." -- https://docs.mlcommons.org/croissant/docs/croissant-spec.html (Version 1.0)

**Comment by kbrueckmann** [comment]
Thank you, @jggautier, for the quick reply and the follow-up!

> [@kbrueckmann](https://github.com/kbrueckmann), could you share how much of a priority this is for you?

This has medium priority for us.

> Do you think it would be possible for you or someone in the Dataverse community to follow those guidelines from "Science on Schema.org" so that grant IDs are included?

Yes, I think so, though it probably won't be me. When we first received your reply we decided to wait for now as you mentioned that there might be changes to the funding metadata fields.

> Do you think Google's validation tools would be okay with those changes?

I can't really say as I'm not working with Google's validation tools. My best guess – but still a guess – would be yes as Google advises to include metadata as fully as possible and MonetaryGrant is a valid schema.org Type.

**Comment by jggautier** [comment]
Ah, I see. Sorry about that. We haven't had the resources to address as many of the challenges with the funding metadata fields that we wrote about in https://github.com/IQSS/dataverse/issues/4859 and https://github.com/IQSS/dataverse/issues/10196. For now I'd say that those challenges shouldn't hold up any work that might be done to add grant IDs to the schema.org export.

@cmbz, since including Grant IDs in the Schema.org exports isn't a priority in light of new work funded by the NIH GREI grant, but is a medium priority for @kbrueckmann, is it okay to keep this open until @kbrueckmann and/or others in the Dataverse community can work on it?

**Comment by pdurbin** [comment]
Just a heads up that it's possible to override an exporter by creating your own with the same internal name (`getFormatName`). Here's how we describe this in the guides:

"If this value is set, Dataverse will examine all JARs in the specified directory and will use them to add new metadata export formats or (if the machine-readable name used in [Export Metadata of a Dataset in Various Formats](https://guides.dataverse.org/en/6.6/api/native-api.html#export-dataset-metadata-api) is the same) replace built-in metatadata export formats."

-- https://guides.dataverse.org/en/6.6/installation/config.html#dataverse-spi-exporters-directory

In this case `getFormatName` would need to be "schema.org".

**Comment by cmbz** [comment]
2025-07-07
- Currently not a GREI priority, however, this is a valuable feature and we'll keep open.

**Comment by gtsueng** [comment]
>Do you think Google's validation tools would be okay with those changes?

For any CreativeWork derivative class, including Datasets, we have used the 'funding' property with a MonetaryGrant value to include both funder and grant IDs. Although the 'funding' property is pending in schema.org, it has not caused issues for the [schema.org/validator](https://validator.schema.org/) nor has it caused issues with the [google rich results test](https://search.google.com/test/rich-results) in our tests. 

* [Example of a Dataset record in the schema.org validator](https://validator.schema.org/#url=https%3A%2F%2Fdata.niaid.nih.gov%2Fresources%3Fid%3Dvivli_89794842-f2ba-41f9-99c5-abd318aec004): There are errors associated with the schema that are primarily due to properties in our schema that don't exist on schema.org; but 'funding' is not any of the errors.
* [Example of a Dataset record in the google rich results test](https://search.google.com/test/rich-results?url=https%3A%2F%2Fdata.niaid.nih.gov%2Fresources%3Fid%3Dvivli_89794842-f2ba-41f9-99c5-abd318aec004): Only has a warning for missing 'license' information

**Comment by cattapre** [comment]
I tried to have the UCSC Genome Browser to access data stored on the Harvard Dataverse, but it cannot. After my colleague working with nice folks at the Dataverse, apparently the problem is that there are redirects that the Genome Browser cannot handle. All possibilities raised by the Harvard Dataverse team have been tried, but none worked. The file access permissions were correct.
So, my request is to implement a feature so that we can use the Genome Browser (and other web apps alike) to access the data stored on the Harvard Dataverse. Files could be large in size (GB range) and multiple of them in a single session, and so for practicality latency time cannot be large. Such track hubs are public available, but not intensively accessed.
Thanks!

**Comment by jggautier** [comment]
Heya @cattapre. I think this GitHub is unrelated to your comment. I think there are other GitHub issues that are more related so I'll try to look for 'em now and see if we can add your comment in one of those.

**Comment by jggautier** [comment]
Hmm, I couldn't find an open GitHub issue that I think is directly related. I think [your comment](https://github.com/IQSS/dataverse/issues/10296#issuecomment-3873103511) is related to the goals and conversations in the GitHub issue at https://github.com/IQSS/dataverse/issues/3584, although that issue is closed.

So maybe it's best to open a new GitHub issue? @pdurbin would you agree?

**Comment by pdurbin** [comment]
Yes, a new GitHub issue would be ideal. @cattapre are you willing to open one? Thanks!

**Comment by pdurbin** [comment]
Just a note that as of this writing (3e5bba0) Croissant pulls in funder from the Schema.org format:

            JsonArray funder = datasetSchemaDotOrg.getJsonArray("funder");
            if (funder != null) {
                job.add("funder", funder);
            }

It looks like this https://github.com/IQSS/dataverse/blob/3e5bba0e7cb6435b37efe6f3bdff19dcd2980346/src/test/resources/croissant/max/expected/max-croissant.json#L97-L106

I'm giving this a size of 30. The main things to figure out seem to be:

- How much divergence do we want between `schema.org` and `croissant` export formats?
- For both formats, we need to ensure that validation tools and are still happy with our choices.
- What does the Croissant spec have to say about this? Do we need any changes in the spec?
- Should we switch to https://schema.org/funding which will probably be a backward incompatible change?

**Comment by pdurbin** [comment]
@kbrueckmann what if we added the grant ID to the Croissant format instead? Would that work for you?

**Comment by pdurbin** [comment]
@DS-INRAE same question. 😄

**Comment by kbrueckmann** [comment]
Our use case was that we received a request aimed at populating/creating a registry based on schema.org metadata. We haven’t heard back from them (yet), which means they may have used the Datacite/OpenAIRE/... metadata export instead—which includes the funder and grant ID—or the priority for this has been downgraded on their side. I still think that it would be beneficial to include as much metadata into the different export options as possible, so I'm happy to keep the issue open, but as there is no urgent need and a lot of other work to be done, we could also close it.

**Comment by pdurbin** [comment]
@kbrueckmann hi! Yesterday I was just arguing that perhaps we should deprecate the "Schema.org JSON-LD" metadata format because it's not base on any spec and was really just a precursor to what evolved into Croissant, which does has a spec.

We initially added it for and with Google as they developed Google Dataset Search. However, these days, they want everyone to use Croissant instead.

Here's how I wrote up the situation in the guides:

"We include Croissant in the <head> because it’s [recommended](https://github.com/mlcommons/croissant/issues/530#issuecomment-1964227662) by Google for [Google Dataset Search](https://datasetsearch.research.google.com/), where they offer a filter to narrow results to only datasets with support for Croissant.

Before Croissant was invented, Google recommended a different format that Dataverse refers to as “Schema.org JSON-LD” in the user interface (and schema.org in the API). If you prefer to put that older format in the `<head>`, which was the behavior in older versions of Dataverse, see [dataverse.legacy.schemaorg-in-html-head](https://guides.dataverse.org/en/6.11/installation/config.html#dataverse-legacy-schemaorg-in-html-head)."

-- https://guides.dataverse.org/en/6.11/admin/discoverability.html#croissant-metadata-in-the-head-of-dataset-landing-pages
